hints/permutation-string.md

```diff
@@ -0,0 +1,31 @@
+<br>
+<details class="hint-accordion">  
+    <summary>Recommended Time & Space Complexity</summary>
+    <p>
+    You should aim for a solution with <code>O(n)</code> time and <code>O(1)</code> space, where <code>n</code> is the maximum of the lengths of the two strings.
+    </p>
+</details>
+
+<br>
+<details class="hint-accordion">  
+    <summary>Hint 1</summary>
+    <p>
+    A brute force solution would be to check every substring of <code>s2</code> with <code>s1</code> by sorting <code>s1</code> as well as the substring of <code>s2</code>. This would be an <code>O(n^2)</code> solution. Can you think of a better way? Maybe we can use the freqency of the characters of both the strings as we did in checking anagrams.
+    </p>
+</details>
+
+<br>
+<details class="hint-accordion">  
+    <summary>Hint 2</summary>
+    <p>
+    It is obvious to return false when the length of <code>s1</code> is greater than the length of <code>s2</code>. We can simply use an array of size <code>O(26)</code>, since the character set is <code>a</code> through <code>z</code> (<code>26</code> continuous characters), to count the frequency of each character in a string. Which algorithm can we use now?
```

Comment:
Minor feedback:

1. Remove "it is obvious"
2. In the second sentence, i would start with "To count the frequency.. we can simply use an array.." Starting with the reason makes it easier to understand. Otherwise when i start that sentence idk why we are using an array.